### PR TITLE
fix: remove evtAction middleware

### PR DESCRIPTION
**Pull request by garronej**

### CODE REVIEW

File: src/app/use-cases/counter.ts [hunk 1]
```diff
@@ -43,11 +43,19 @@ export const selectors = {
...
+
+    return {
+      [thunkName]: createAsyncThunk<number, number, AsyncThunkConfig>(`${name}/${thunkName}`, async (amount: number, { extra }) => {
+        const { data } = await extra.counterApi.fetchCount(amount);
+
+        return data;
+      })
+    };
+
+  })(),
```
Comment (garronej) [inline src/app/use-cases/counter.ts]: ...This is why I no longer use `createAsyncThunk`. I don't think there is another way to write it that do not requires to keep tokens in sync.

Comment (leosuncin) [reply src/app/use-cases/counter.ts]: There isn't a problem when the action name ('counter/fetchCount') is not the same as the function name ('incrementAsync'), to be continued.

File: src/app/use-cases/counter.ts [hunk 2]
```diff
@@ -43,19 +43,11 @@ export const selectors = {
 };
 
 export const thunks = {
-  ...(() => {
+  incrementAsync: createAsyncThunk<number, number, AsyncThunkConfig>(`${name}/incrementAsync`, async (amount: number, { extra }) => {
```
Comment (garronej) [inline src/app/use-cases/counter.ts]:
```diff
- incrementAsync: createAsyncThunk<number, number, AsyncThunkConfig>(`${name}/incrementAsync`, async (amount: number, { extra }) => {
+ incrementAsync: createAsyncThunk<number, number, AsyncThunkConfig>(`${name}/incrementAsync`, async (amount, { extra }) => {
```
Here, we don't need type annotaition for the type of `ammount` it's constrained by the second type argument of `createAsyncThunk`

File: src/app/use-cases/counter.ts (hunk 1 above)
Comment (garronej) [reply src/app/use-cases/counter.ts]: > There isn't a problem when the action name ('counter/fetchCount')

You are right. I still feel that they should be matching though...

**Pull request merged**